pkg/terraform/store.go

```diff
@@ -121,6 +133,13 @@ func (ws *WorkspaceStore) Workspace(ctx context.Context, c resource.SecretClient
 		return nil, errors.Wrap(err, "cannot write main tf file")
 	}
 	l := ws.logger.WithValues("workspace", dir)
+	reattachConfig, err := ws.providerRunner.StartSharedServer()
```

Comment:
I got to learn what this might return from our offline chat. `attachmentConfig` could be less specific so that others understand what it is immediately. Or maybe `terraformEnvConfig` that we can set directly.